ViewPumpSample/src/main/java/io/github/inflationx/viewpump/sample/MyViewInterceptor.java

```diff
@@ -0,0 +1,38 @@
+package io.github.inflationx.viewpump.sample;
+
+import android.content.Context;
+import android.support.annotation.Nullable;
+import android.util.AttributeSet;
+import android.view.View;
+
+import io.github.inflationx.viewpump.InflateRequest;
+import io.github.inflationx.viewpump.InflateResult;
+import io.github.inflationx.viewpump.Interceptor;
+
+public class MyViewInterceptor implements Interceptor {
```

Comment:
the flow is analogous to a network request, where the inflation of the view and view is returned is the request hitting the network and returning a response. a pre-inflation interceptor would be returning a new instance of a programmatically instantiated view instead of inflating a view, much like returning a result from the network cache instead of hitting the network. a post-inflation interceptor would be doing something with the inflated view like applying a typeface, much like a network response that can have its output logged to the console.

i'm all for making it easier to use, let me think on this a bit more. let me know if the above explanation changes your opinion